.github/workflows/github-cxx-qt-tests.yml

```diff
@@ -88,6 +88,55 @@ jobs:
           token: ${{ secrets.CODECOV_TOKEN }}
           verbose: true
 
+  build-wasm:
+    needs: [clang_format, license_check, rust_format_check, markdown_lint, shellcheck]
+    runs-on: ubuntu-22.04
+    env:
+      SCCACHE_CACHE_SIZE: 600M
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+      - name: Setup toolchain
+        run: |
+          rustup component add clippy rustfmt
+          rustup target add wasm32-unknown-emscripten
+          git clone https://github.com/emscripten-core/emsdk
+          ./emsdk/emsdk install 2.0.14
+          ./emsdk/emsdk activate 2.0.14
+          source ./emsdk/emsdk_env.sh
```

Comment:
This could be moved into it's own step of `Setup emsdk`.

Then see below about `Rust tool cache` + `Build rust tools` as you could copy this and use a cache with a key for `emsdk_2.0.14` and cache the sdk paths to save another 30s.

Also note that the `source` here is probably pointless as we found that each step has it's own env?